app/views/proposals/show.html.erb

```diff
@@ -9,12 +9,16 @@
   <%= render @comment.proposal %>
 
   <% if user_signed_in? %>
-    <%= render 'comments/form' %>
+    <div class="content-box <%= @comments.any? ? 'top' : ''%>">
```

Comment:
same question here.. will `any` check in the array or will it trigger a `count` query?

Reply:
No idea. Why the concern? Any guess how to break it?

Reply:
let me explain:

when `@comments` is loaded: https://github.com/call4paperz/call4paperz/blob/master/app/controllers/proposals_controller.rb#L11 actually is lazy loaded.

when `@comments.any?` is called, a count query will be executed and then another query when `.each_with_index` is called. 

basically `.any?` will generate an unnecessary count query (there is also an `any` method on active record other than ruby arrays).

it can be easily fixed calling `.to_a` on https://github.com/call4paperz/call4paperz/blob/master/app/controllers/proposals_controller.rb#L11 (no more lazy loaded)

Reply:
I've tested you suggestion, but according to the console, it's still doing the same queries.

Reply:
have you seen a query like that ` SELECT COUNT(*) FROM "comments" WHERE "comments"."proposal_id" = $1`? if so, it might be avoided

Reply:
Now I got it. I wasn't seeing because I was testing in the events#show. My bad. I've updated the code. Take a look if that's enough or do you have a better idea.

Reply:
👍